editor/src/components/editor/actions/actions.tsx

```diff
@@ -1082,10 +1087,12 @@ function deleteElements(targets: ElementPath[], editor: EditorModel): EditorMode
 
       function deleteElementFromParseSuccess(parseSuccess: ParseSuccess): ParseSuccess {
         const utopiaComponents = getUtopiaJSXComponentsFromSuccess(parseSuccess)
-        const withTargetRemoved: Array<UtopiaJSXComponent> = removeElementAtPath(
-          targetPath,
-          utopiaComponents,
-        )
+        const { components: withTargetRemoved, affectedElements: newAffectedElements } =
+          removeElementAtPath(targetPath, utopiaComponents)
+        affectedElements = {
```

Comment:
Maybe it is just my preference, and feel free to ignore it, but I dislike using reduce with a function which has an important side effect. In this case I would just use a for loop or a forEach
(My philosophy is that a reduce should continue to work well even if you memoize the function inside it).

Reply:
something like this? https://github.com/concrete-utopia/utopia/pull/3482/commits/b264e393bbf76db93a22bf1c4bc5ad988f440f5e

Reply:
Absolutely agree, reduce shouldn't be triggering side effects. The function could be modified to return an object containing both the `updatedEditor` and the `affectedElements` instead if we wanted to do that.